cunumeric/deferred.py

```diff
@@ -651,6 +677,8 @@ def _create_indexing_array(
             # the store with transformation
             rhs = cast(DeferredArray, self._copy_store(store))
 
+        if len(tuple_of_arrays) == 0:
+            return False, rhs, rhs, self
```

Comment:
I don't think this can ever happen anymore, since the special case was removed above. We could instead just
```
assert(len(len_tuple_of_arrays) > 0)
```
to make this clear on the code.

Reply:
probably not. removing this case